bluetooth/src/commonTest/kotlin/device/DeviceTest.kt

```diff
@@ -304,21 +305,25 @@ class DeviceTest : BluetoothFlowTest<DeviceState>() {
         connectionManager.waitAfterHandlingAction[DeviceAction.Read.Characteristic::class] = EmptyCompletableDeferred()
         connectionManager.waitAfterHandlingAction[DeviceAction.Write.Descriptor::class] = EmptyCompletableDeferred()
 
+        val readAction = DeviceAction.Read.Characteristic(characteristic)
```

Comment:
Read and write action are defined too early. They're only used within the when statement, so why create them before we even do the state transition

Reply:
Good point